Add more sample name options

troycomi opened the pull request:
- Added an argument to get_samplename_list so samples can be matched
  with population, superpopulation or sample.
- Added options to specify modern super population and archaic sample
- Changed logic of parser so none of those are required.  Instead,
  check if either list is empty and raise an exception if so indicating
  which options should be specified.

troycomi committed:
Add more sample name options

- Added an argument to get_samplename_list so samples can be matched
  with population, superpopulation or sample.
- Added options to specify modern super population and archaic sample
- Changed logic of parser so none of those are required.  Instead,
  check if either list is empty and raise an exception if so indicating
  which options should be specified.

lparsons reviewed (changes requested):
Thanks for the addition, it looks good to me!

Before we merge, could we add a note to the changelog and bump the version number?

troycomi commented:
Will add in the next push.  It will be a few days before the empirical data is run through.

troycomi committed:
Add query-on option

- Added option to match superpopulation of query to population of the
  null database.  Necessary for current simulation framework.
- Bumped version number
- Updated changelog
